Remove hard-coded scheme in azure HTTP requests. Closes #1703

KatiRG opened the pull request:
## Dynamically assign scheme to HTTP requests in azure driver

### Description
Per #1703. The hard-coded `http` scheme is replaced with a variable that assigns it to `http` or `https` based on the value of `self.secure`. Otherwise, requests made to a secure Azure blob storage will always start with `http` and will fail, as this is not allowed.

### Status

- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

Output of `tox`:
```
========================================== 25 passed, 39 skipped in 13.11s ===========================================
______________________________________________________ summary _______________________________________________________
ERROR:  pypypy3: InterpreterNotFound: pypy3
ERROR:  py3.6: InterpreterNotFound: python3.6
ERROR:  py3.7: InterpreterNotFound: python3.7
ERROR:   py3.8: could not install deps [-r/home/ckan29lts/FORKS/libcloud/requirements-tests.txt, fasteners, libvirt-python==7.9.0]; v = InvocationError('/home/ckan29lts/FORKS/libcloud/.tox/py3.8/bin/python -m pip install -r/home/ckan29lts/FORKS/libcloud/requirements-tests.txt fasteners libvirt-python==7.9.0', 1)
ERROR:  py3.9: InterpreterNotFound: python3.9
ERROR:  py3.10: InterpreterNotFound: python3.10
ERROR:  pypyjion: InterpreterNotFound: pyjion
ERROR:  checks: InterpreterNotFound: python3.7
ERROR:  lint: InterpreterNotFound: python3.7
ERROR:  pylint: InterpreterNotFound: python3.7
ERROR:  mypy: InterpreterNotFound: python3.7
ERROR:  docs: InterpreterNotFound: python3.7
ERROR:  coverage: InterpreterNotFound: python3.7
  integration-storage: commands succeeded
```

KatiRG committed:
Remove hard-coded scheme in azure HTTP requests. Closes #1703

aimalkhan reviewed (approved):
We have been having issues with this module and incorporating these changes upstream will be helpful.

Kami committed:
Try to re-enable pypy job.

Kami committed:
Update more checks.

Kami committed:
Merge branch 'assign_scheme_in_azure_driver' of https://github.com/KatiRG/libcloud into KatiRG-assign_scheme_in_azure_driver

Kami committed:
Fix lint.

Kami committed:
Add a test case for it.

Kami committed:
Fix a bug in Azure Blobs storage driver "list_containers()" method and
make sure unncessary / redundant double quotes are stripped from the
Container extra["etag"] attribute.

Kami committed:
Add changelog entry.

Kami committed:
Fix daily install job.

Kami commented:
I added a test case for it (e87ee8314e8a88ee908d109ac877dc3f36046e7d) and will go ahead and merge it into trunk. Thanks for the contribution.

While testing the fix and adding a test case, I also noticed a bug in the ``list_containers()`` method where double quotes are not correctly removed from the ``etag`` extra attribute.

```bash
......{'url': None, 'last_modified': None, 'etag': '"0x8CFBAB7B5B82D8E"', 'lease': {'status': 'unlocked', 'state': 'available', 'duration': None}, 'meta_data': {}}
```

I went ahead and fixed that as well - 07c72cd8ec8ddea0636d9632a72c66e5d6975777.
